### ISSUE TITLE
Gadfly embedded NULs and invalid plotting

### ISSUE THREAD

**Issue by datadreamscorp** [GiovineItalia/Gadfly.jl#1535]
Opened one of my scripts today to find this error whenever I tried to plot using a DataFrame in Pluto.jl. Tried plotting to vectors in another session and axis ticks are not showing properly (image below), which I guess is part of the problem.

Failed to show value:

ArgumentError: embedded NULs are not allowed in C strings: "-\0.\0"

    unsafe_convert@c.jl:216[inlined]
    pango_to_svg(::String)@pango.jl:330
    draw(::Compose.SVG, ::Compose.TextPrimitive{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}, Compose.Rotation{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}}, Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}}, ::Int64)@svg.jl:975
    draw(::Compose.SVG, ::Compose.Form{Compose.TextPrimitive{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}, Compose.Rotation{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}}, Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}}})@svg.jl:732
    drawpart(::Compose.SVG, ::Compose.Context, ::Compose.IdentityTransform, ::Compose.UnitBox{Float64, Float64, Float64, Float64}, ::Measures.BoundingBox{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}, Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}})@container.jl:477
    drawpart(::Compose.SVG, ::Compose.Context, ::Compose.IdentityTransform, ::Compose.UnitBox{Float64, Float64, Float64, Float64}, ::Measures.BoundingBox{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}, Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}})@container.jl:509
    drawpart(::Compose.SVG, ::Compose.AdhocContainerPromise, ::Compose.IdentityTransform, ::Compose.UnitBox{Float64, Float64, Float64, Float64}, ::Measures.BoundingBox{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}, Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}})@container.jl:390
    drawpart(::Compose.SVG, ::Compose.Context, ::Compose.IdentityTransform, ::Compose.UnitBox{Float64, Float64, Float64, Float64}, ::Measures.BoundingBox{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}, Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}})@container.jl:509
    drawpart(::Compose.SVG, ::Compose.Table, ::Compose.IdentityTransform, ::Compose.UnitBox{Float64, Float64, Float64, Float64}, ::Measures.BoundingBox{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}, Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}})@container.jl:390
    drawpart(::Compose.SVG, ::Compose.Context, ::Compose.IdentityTransform, ::Compose.UnitBox{Float64, Float64, Float64, Float64}, ::Measures.BoundingBox{Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}, Tuple{Measures.AbsoluteLength, Measures.AbsoluteLength}})@container.jl:509
    draw@container.jl:356[inlined]
    show(::IOContext{IOBuffer}, ::MIME{Symbol("text/html")}, ::Compose.Context)@Compose.jl:182
    show_richest(::IOContext{IOBuffer}, ::Any)@PlutoRunner.jl:640
    var"#sprint_withreturned#28"(::IOContext{Base.DevNull}, ::Int64, ::typeof(Main.PlutoRunner.sprint_withreturned), ::Function, ::Compose.Context)@PlutoRunner.jl:579
    format_output_default(::Any, ::Any)@PlutoRunner.jl:503
    #format_output#17@PlutoRunner.jl:520[inlined]
    formatted_result_of(::Base.UUID, ::Bool, ::Nothing)@PlutoRunner.jl:435
    top-level scope@none:1

![Screenshot_20210428_113615](https://user-images.githubusercontent.com/59768634/116456176-cdc63d00-a816-11eb-95a0-8f12ae123430.png)

**Comment by datadreamscorp** [comment]
Happening in REPL as well.

**Comment by pdeffebach** [comment]
I can reproduce on Mac with 1.6.0.

**Comment by rikhuijzer** [comment]
It also occurs for me on NixOS with a fresh Julia after updating my dependencies. I'll see whether I can find the upstream culprit.

**Comment by Mattriks** [comment]
There's a [discourse thread](https://discourse.julialang.org/t/plotting-broken-gadfly-and-plots-backends-not-showing-axis-ticks-cant-plot-dataframes/60201) too

**Comment by rikhuijzer** [comment]
When looking at the dependencies, I would expect it to be [Colors.jl](https://github.com/JuliaGraphics/Colors.jl) since it got updated two days ago. Unfortunately, pinning that to `0.12.6` as used [here](https://github.com/GiovineItalia/Compose.jl/runs/1898123783) doesn't seem to be helping.

**Comment by rikhuijzer** [comment]
I've got the first Gadfly error on 2021-04-28 around 15:00 GMT. When looking through the General registry for changes, there recently have been updates to:

- [Colors.jl](https://github.com/JuliaGraphics/Colors.jl)
- [ColorTypes.jl](https://github.com/JuliaGraphics/ColorTypes.jl)
- [StaticArrays.jl](https://github.com/JuliaArrays/StaticArrays.jl)

I've pinned them all to older versions with a Gadfly `dev` project, but plots keep failing. I guess that to fix this, it is necessary to approach the problem from another direction, namely debugging the code.
